### PR TITLE
feat(notifier): support group topics in Telegram sender

### LINKED ISSUE #1009: Add the ability to send notifications to telegram topics
# FEATURE

## Summary
Now Moira supports telegram sender, which knows how to send notifications to personal, private, public channels. We need to add the ability to specify specific channel topics

### Research
As one option, add another way to enter a custom value in contact and on backend parse the channel id and a specific topic from it

Need to see how to send notifications by topic in the Telegram client we are working with

### PR DESCRIPTION
**Pull request by almostinf**
# Support group topics in Telegram sender

This PR is for the most part a copy of the PR #1022 (Authored by: @Gaspero) with a few changes:
1. Kept backward compatibility in the `getChatFromDb` function, so as not to disable Moira for migrations. The code can correctly handle keys like `moira-telegram-users:test_user`:`-1001231231` (old version) and `moira-telegram-users:test_user`:`{"chat_id": "-1001231231"}` (new version)
2. Fixed some issues in migration from 2.11 -> 2.12: added rollup of changes to all redis shards + bug with `moira-telegram-users:moira-bot-host:123` key
3. Changed the library for mocks from https://github.com/golang/mock to https://github.com/uber-go/mock and fixed all dependencies
4. Removed the `type` field from the `Chat` structure in the telegram sender
5. Improved naming of functions for working with bot in Redis

Closes #1009

### CODE REVIEW

File: cmd/cli/main.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ var (
 	GoVersion    = "unknown"
 )
 
-var moiraValidVersions = []string{"2.3", "2.6", "2.7", "2.9"}
+var moiraValidVersions = []string{"2.3", "2.6", "2.7", "2.9", "2.11"}
```
Comment (Tetrergeru) [inline cmd/cli/main.go:28]: Может это должно быть `2.12`?

Comment (almostinf) [reply cmd/cli/main.go:28]: нет, обновление `--update --from-version 2.11`, даунгрейд `--downgrade --to-version 2.11`

**Review by Tetrergeru — APPROVED** [review]

File: cmd/cli/main.go (hunk 1 above)
Comment (Tetrergeru) [reply cmd/cli/main.go:28]: Ладно, справедливо (удобно, конечно)

**Review by Tetrergeru — APPROVED** [review]

**Pull request merged**